### ISSUE TITLE
matches might return nil

### ISSUE THREAD

**Issue by bodrovis** [vinnicc/dota#10]
Hello once again :)

I've received some feedback on my article and some users were getting 500 error. I've browsed through the logs and seems like the error was here https://github.com/bodrovis/Sitepoint-source/commit/49fc79984275055e87ed38c6ae53ac6790e97ef9#diff-38afbfd893e780c8001bb0d8998adcd0L40 So as you see I am just iterating over an array of matches. However sometimes `matches` return `nil` and therefore calling `each` on it made the app crash.

So basically the `matches` method https://github.com/vinnicc/dota/blob/8f12d56444c0bed3bba9f3dd6d46898eec8b5352/lib/dota/api/client.rb#L39 really can return `nil` (because the user has no matches?) - is this an expected behavior? Wouldn't it be better to return an empty array instead?

**Comment by vinnicc** [comment]
Yes that would be better. I assumed Steam will always return an array and in case `response["matches"]` returns `nil`, i want it to break since it means the API's schema has been changed or for some reason I'm calling an incorrect endpoint. But as things are right now, that's a bug and it should fall back to an empty array since you're expecting a list.

Doing `(matches = response["matches"] || [])` would solve the issue. This applies for the other endpoints as well.

**Comment by bodrovis** [comment]
Yeah, I'll send PR then. This happens when user's account is marked is private and therefore no matches can be loaded.
